docs/guidelines.md

```diff
@@ -0,0 +1,201 @@
+# Actions Guidelines
+
+> Last updated: **April 24th, 2025**
+
+This document outlines a shared set of guidelines for creating and maintaining GitHub Actions files for Hemi Labs.
+
+_These guidelines are currently a work-in-progress proposal and need review and agreement from all engineering teams at
+Hemi Labs._
+
+---
+
+## 1. Reusable actions and workflows
+
+The [`hemilabs/actions`](https://github.com/hemilabs/actions/) repository is the primary repository for GitHub Actions
+and workflows.
+
+- Workflows or actions used in multiple repositories should be placed here, unless there's a specific reason not
+  to (e.g., internal, or environment-specific logic).
+- Aim to make actions and workflows configurable and well-documented to allow reuse.
+- Breaking changes to shared actions or workflows should be discussed prior to being made to avoid unexpected issues in
+  consuming projects.
+
+### 1.1. Versioning
+
+The `hemilabs/actions` repository uses tagged releases, following Semantic Versioning. Consumers of actions or workflows
+from `hemilabs/actions` should pin the action/workflows to a `vX.X` or `vX.X.X` tag to avoid unexpected or unreviewed
+changes.
```

Comment:
```suggestion
from `hemilabs/actions` should pin the action/workflows to a `vX.X` or `vX.X.X` tag to avoid unexpected or unreviewed changes.
```

It seems `changes` here is in a new line?

Reply:
Oops, must have happened during formatting. Thanks!